### PR TITLE
feat(popover): Add popover web component

### PR DESCRIPTION
**Pull request by miroslavstastny**
WIP

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/237267b5ad816d090011c93f2922476c1abfe87f'>237267b5ad816d090011c93f2922476c1abfe87f</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/web-components/src/popover/popover.styles.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { css } from '@microsoft/fast-element';
+import * as tokens from '../theme/design-tokens.js';
+
+export const styles = css`
+  :host {
+    display: contents;
+  }
+
+  .popover-content ::slotted(*) {
```
Comment (eljefe223) [inline packages/web-components/src/popover/popover.styles.ts:9]: You can just use the `::slotted` selector here.

File: packages/web-components/src/popover/popover.ts [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+  protected openChanged() {
+    console.group(this.objId, 'openChanged');
+
+    // `open` has been set to true, anchor and popover element might already be in DOM (perhaps subsequent open)
+    this.handlePositioningStartStop();
+    console.groupEnd();
+  }
+
+  @observable
+  public popoverContentRef?: FASTElement; // @FIXME: what type?
```
Comment (eljefe223) [inline packages/web-components/src/popover/popover.ts:34]: `HTMLElement | undefined`

File: packages/web-components/src/popover/popover.ts [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+      open: this.open,
+      connected: (this as FASTElement).$fastController.isConnected,
+      anchorRef: this.anchorRef,
+      popoverContentRef: this.popoverContentRef,
+      shouldStart,
+      positioningStarted: this.isPositioningStarted,
+      forceRestart,
+    });
+
+    if (shouldStart) {
```
Comment (eljefe223) [inline packages/web-components/src/popover/popover.ts:140]: Do you need this check it seems that either way ` if (this.isPositioningStarted)` then you call `this.stopPositioning`

File: packages/web-components/src/popover/popover.ts [hunk 4]
```diff
@@ -0,0 +1,292 @@
...
+      this.startPositioning();
+    } else {
+      if (this.isPositioningStarted) {
+        this.stopPositioning();
+      }
+    }
+  }
+
+  private startPositioning() {
+    console.log(this.objId, 'startPositioning', {
```
Comment (eljefe223) [inline packages/web-components/src/popover/popover.ts:157]: We need to clean up all these console.logs

File: packages/web-components/src/popover/popover.ts [hunk 5]
```diff
@@ -0,0 +1,292 @@
+import { attr, FASTElement, observable, Updates } from '@microsoft/fast-element';
+import { autoUpdate, computePosition } from '@floating-ui/dom';
+import { PopoverAlignment, PopoverPosition } from './popover.options.js';
+import { toFloatingUIPlacement } from './toFloatingUIPlacement.js';
+
+let nextEmoji = 0;
+const getNextEmoji = () => {
+  // prettier-ignore
+  const emojis = ['🍇', '🍉', '🍊', '🍋', '🍌', '🍍', '🥭', '🍎', '🍏', '🍐', '🍑', '🍒', '🍓', '🫐', '🥝', '🍅', '🫒', '🥥', '🥑', '🍆', '🥔', '🥕', '🌽', '🌶', '🫑', '🧄', '🥨', '🚌'];
```
Comment (eljefe223) [inline packages/web-components/src/popover/popover.ts:9]: Whats happening here with the emojis? Are these supposed to be baked in?

**Comment by layershifter** [comment]
Closing for housekeeping 🏡

**Pull request closed without merging**